app/src/main/java/com/cmput301f20t21/bookfriends/ui/request/RequestActivity.java

```diff
@@ -0,0 +1,177 @@
+        // bind the text view
+        titleTextView = findViewById(R.id.title_text_view);
+        authorTextView = findViewById(R.id.author_text_view);
+        descriptionTextView = findViewById(R.id.description_text_view);
+        bookStatus = findViewById(R.id.status_text_view);
+        bookImage = findViewById(R.id.book_image_view);
+
+        requestViewModel = new ViewModelProvider(this).get(RequestViewModel.class);
+
+        // since the data is mutable live, set the observer so the content will change accordingly
+        requestViewModel.getBook(bookId).observe(this, book -> {
+            titleTextView.setText(book.getTitle());
+            authorTextView.setText(book.getAuthor());
+            descriptionTextView.setText(book.getDescription());
+            bookStatus.setText(book.getBookStatus().toString());
+        });
+
+        // also getting the image
+        requestViewModel.getImageUri().observe(this, uri -> {
+            Glide.with(this).load(uri).into(bookImage);
+        });
+    }
+
+    /**
+     * display requesters of the book by getting them from FireStore based on bookId
+     * @param bookId
+     */
+    public void displayRequest(String bookId) {
+        requestViewModel.getRequesters(bookId).observe(this, requesters -> {
+            if (requesters != null) {
+                requestDataList.addAll(requesters);
+                requestAdapter.notifyDataSetChanged();
+            }
+        });
+    }
+
+    /**
+     * setup the back button on the title
+     * @param item
+     * @return
+     */
+    @Override
+    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
+        if (item.getItemId() == android.R.id.home) {
+            finish();
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    /**
+     * Function to set view by ID, set adapter and build recycler view
+     */
+    public void buildRecyclerView() {
+        recyclerView = findViewById(R.id.request_recycler_view);
+        recyclerView.setHasFixedSize(true);
+        layoutManager = new LinearLayoutManager(this);
+        recyclerView.setLayoutManager(layoutManager);
+        requestAdapter = new RequestAdapter(requestDataList);
+        recyclerView.setAdapter(requestAdapter);
+
+        requestAdapter.setOnItemClickLisener(new RequestAdapter.OnItemClickListener() {
+            @Override
+            public void onRejectClick(int position) {
+                removeItem(position);
+            }
+
+            @Override
+            public void onAcceptClick(int position) {
+                openDialog(position);
+            }
+        });
+    }
+
+    /**
+     * function to remove an item when we click on Reject button
+     * @param position that needs removing
+     */
+    public void removeItem(int position) {
+        Request request = requestDataList.get(position);
+        requestViewModel.removeRequest(request.getId());
+        requestDataList.remove(position);
```

Comment:
As mentioned above, are we doing the remove twice here? one in view model and one in the local list.